core/src/main/scala/org/apache/spark/deploy/DeployMessage.scala

```diff
@@ -108,6 +108,9 @@ private[deploy] object DeployMessages {
 
   case class Heartbeat(workerId: String, worker: RpcEndpointRef) extends DeployMessage
 
+  // Out of band commands to Master
+  case class DecommissionHostPorts(hostPorts: Seq[String])
```

Comment:
nit: maybe `DecommissionWorkers`? In the comment, we can say that the worker is identified by host and port(optional)

Reply:
It would be confusing to say `DecommissionWorkers` but passes in sequence of hostPorts...

Reply:
but `DecommissionHostPorts` is more confusing as you don't even know what it does unless you look at the comment.

Reply:
let's change to `DecommissionWorkers` then, since `WorkerStateResponse` also passes in `host` and `port`.

Reply:
I am narrowing the scope to simply decommission a bunch of hostnames, and all workers within a host will go away. This is the only production use case I have in mind and there is no need to design for the flexibility of wanting to decommission an individual worker on the node.

As such, I have renamed the API to `DecommissionHosts` and it takes a list of host names.